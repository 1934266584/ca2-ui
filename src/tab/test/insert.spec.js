import { mount, later } from '../../../test';

// this case will throw wierd error in index.spec.js
// so separate it
test('insert tab dynamically', async () => {
  const wrapper = mount({
    template: `
      <zv-tabs v-model="active">
        <zv-tab title="1">1</zv-tab>
        <div v-if="insert">
          <zv-tab title="2">2</zv-tab>
        </div>
        <zv-tab title="3">3</zv-tab>
      </zv-tabs>
    `,
    data() {
      return {
        insert: false,
        active: 1,
      };
    },
  });

  await later();
  wrapper.setData({ insert: true });
  expect(wrapper).toMatchSnapshot();
});

// this case will throw wierd error in index.spec.js
// so separate it
test('insert tab with child component', async () => {
  const wrapper = mount({
    template: `
      <zv-tabs v-model="active">
        <zv-tab title="1">1</zv-tab>
        <my-tab />
        <zv-tab title="3">3</zv-tab>
      </zv-tabs>
    `,
    components: {
      'my-tab': {
        template: `<zv-tab title="2">2</zv-tab>`,
      },
    },
  });

  await later();
  expect(wrapper).toMatchSnapshot();
});
