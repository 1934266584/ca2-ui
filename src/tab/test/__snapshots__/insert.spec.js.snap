// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`insert tab dynamically 1`] = `
<div class="zv-tabs zv-tabs--line">
  <div class="zv-tabs__wrap zv-hairline--top-bottom">
    <div role="tablist" class="zv-tabs__nav zv-tabs__nav--line">
      <div role="tab" class="zv-tab zv-ellipsis"><span class="zv-tab__text">1<!----></span></div>
      <div role="tab" aria-selected="true" class="zv-tab zv-tab--active zv-ellipsis"><span class="zv-tab__text">2<!----></span></div>
      <div role="tab" class="zv-tab zv-ellipsis"><span class="zv-tab__text">3<!----></span></div>
      <div class="zv-tabs__line" style="width: 0px; transform: translateX(0px) translateX(-50%);"></div>
    </div>
  </div>
  <div class="zv-tabs__content">
    <div role="tabpanel" class="zv-tab__pane" style="display: none;">
      <!---->
    </div>
    <div>
      <div role="tabpanel" class="zv-tab__pane">2</div>
    </div>
    <div role="tabpanel" class="zv-tab__pane" style="display: none;">3</div>
  </div>
</div>
`;

exports[`insert tab with child component 1`] = `
<div class="zv-tabs zv-tabs--line">
  <div class="zv-tabs__wrap zv-hairline--top-bottom">
    <div role="tablist" class="zv-tabs__nav zv-tabs__nav--line">
      <div role="tab" aria-selected="true" class="zv-tab zv-tab--active zv-ellipsis"><span class="zv-tab__text">1<!----></span></div>
      <div role="tab" class="zv-tab zv-ellipsis"><span class="zv-tab__text">2<!----></span></div>
      <div role="tab" class="zv-tab zv-ellipsis"><span class="zv-tab__text">3<!----></span></div>
      <div class="zv-tabs__line"></div>
    </div>
  </div>
  <div class="zv-tabs__content">
    <div role="tabpanel" class="zv-tab__pane">1</div>
    <div role="tabpanel" class="zv-tab__pane" style="display: none;">
      <!---->
    </div>
    <div role="tabpanel" class="zv-tab__pane" style="display: none;">
      <!---->
    </div>
  </div>
</div>
`;
